databases/tiledb/Portfile

```diff
@@ -0,0 +1,122 @@
+homepage            https://tiledb.com
+checksums           rmd160  38c2a9cd08aa9511354aef0b110974c92c5bad00 \
+                    sha256  2a7785422290b8a0ec4a957bd5e10e0780d954a56f6478509b71d75444dd4b1d \
+                    size    4268969
+
+# Notice that 32-bit platforms are not supported by upstream, and there are reports elsewhere
+# that some targets are broken or fail tests. It may or may not work at any given moment,
+# but seems to require substantial effort to have it properly fixed.
+# ppc64 works much better than ppc32, though some tests fail.
+# I do not disable 32-bit archs in order to facilitate testing and possible fix-ups.
+# Do not expect it to work as-is. See also: https://github.com/TileDB-Inc/TileDB/issues/4096
+
+cmake.module_path-append \
+                    ${prefix}/lib/libfmt10/cmake
+
+depends_lib-append  port:blosc \
+                    port:bzip2 \
+                    port:curl \
+                    port:libfmt10 \
+                    port:libpng \
+                    port:lz4 \
+                    port:spdlog \
+                    port:zlib \
+                    port:zstd
+
+depends_build-append \
+                    port:gmake
+
+if {${os.platform} eq "darwin" && ${os.major} < 13} {
+    # Lion+ (with Xcode 4.1+) have git; earlier need to bring their own.
+    # On 10.8.5 git fetch fails with ssl error.
+    depends_build-append    port:git
+    git.cmd                 ${prefix}/bin/git
+}
+
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    # This trickery is intended. Normal passing of flags to CMake fails to have a desired effect.
+    # Non-Apple OS on ppc will need libatomic linking too, but leave it until tested.
+    patchfiles-append \
+                    patch-linking.diff \
+                    0001-Add-a-fix-for-32-bit.patch
+
+    post-patch {
+        reinplace "s,@PREFIX@,${prefix},g" ${worksrcpath}/cmake/Modules/FindMagic_EP.cmake
+    }
+}
+
+compiler.cxx_standard   2017
+
+# config modules are defunct.
+# https://github.com/TileDB-Inc/TileDB/issues/4070
+configure.args-append \
+                    -DCATCH_INCLUDE_DIR:STRING=${prefix}/include \
+                    -DTILEDB_ABSEIL=OFF \
+                    -DTILEDB_ASSERIONS=OFF \
+                    -DTILEDB_AZURE=OFF \
+                    -DTILEDB_CPP_API=ON \
+                    -DTILEDB_EXPERIMENTAL_FEATURES=OFF \
+                    -DTILEDB_SERIALIZATION=OFF \
+                    -DTILEDB_TESTS=OFF \
+                    -DTILEDB_TOOLS=OFF \
+                    -DTILEDB_VERBOSE=ON \
+                    -DTILEDB_WEBP=OFF \
+                    -DTILEDB_WERROR=OFF
+
+configure.cflags-append \
+                    -DSPDLOG_FMT_EXTERNAL=1
+
+configure.ldflags-append \
+                    -L${prefix}/lib/libfmt10 -lfmt
+
+platform darwin {
+    if {${os.major} < 11 && [string match *gcc* ${configure.compiler}]} {
+        configure.args-append \
+                    -DCOMPILER_SUPPORTS_AVX2=OFF
```

Comment:
gcc building on Intel will be able to use AVX2 ... but it requires MacPorts cctools and a reasonably recently clang to be installed for it to assemble, so I guess just switching it off is the easier path.

Reply:
Here it gets turned off only for building with GCC (which I tried on 10.6.8, and had to add this to fix the build). Clang should be fine.

Reply:
gcc building on Intel will be able to use AVX2 ... but it requires MacPorts cctools and a reasonably recently clang to be installed for it to assemble, so I guess just switching it off is the easier path.

Reply:
Got it, thank you for clarifying. By the way, how does it work together with clang for assembling? Never tried. Will it work automatically once Clang/LLVM is installed?